### DISCUSSION TITLE
blackboard generic set

### DISCUSSION THREAD

**Discussion by bbdb68** [BehaviorTree/BehaviorTree.CPP#661] [Q&A]
Hi, I am new to BTCPP, so sorry if the question has already been discussed.

I would like to integrate BTCPP in the Flecs ECS https://www.flecs.dev/flecs/#/.  In order to achieve this, I need to populate the tree blackboard with Flecs components in order to forward the Flecs context to BTCPP. On the flecs side,  I have type ids (mostly full qualified class names) and access to (untyped) pointer to data. If necessary I probably can convert the components to std::any. 

What is missing (from what I have seen) in the picture is a generic set method on the blackboard. Or is there a way to obtain a writable reference through getAny ?

Thanks for your help.

Bruno

**Comment by facontidavide** [comment]
The blackboard is in fact a map of std::any and can contain any type.

The blackboard should be used for dataflow between nodes, not to share data from the "backend".

Also, if ALL your nodes need to access the same objects, you may want to consider this https://www.behaviortree.dev/docs/tutorial-basics/tutorial_08_additional_args

**Reply by bbdb68** [reply]
thanks for the quick answer.  in fact I am not exactly in the case of this tutorial, in particular the backend data may be modified by the tree, that's why my idea was to fill the blackboard, run the tree and then fetch the blackboard again.  I'll try to formulate this with additional args, maybe I'll need a post execution visitor to fetch the results.
